src/commands/touch.rs

```diff
@@ -0,0 +1,55 @@
+use crate::prelude::*;
+use nu_errors::ShellError;
+use nu_protocol::{CallInfo, Signature, SyntaxShape, Value};
+use nu_source::Tagged;
+use std::error::Error;
+use std::fs::OpenOptions;
+use std::path::PathBuf;
+
+pub struct Touch;
+
+#[derive(Deserialize)]
+pub struct TouchArgs {
+    pub target: Tagged<PathBuf>,
+}
+
+impl PerItemCommand for Touch {
+    fn name(&self) -> &str {
+        "touch"
+    }
+    fn signature(&self) -> Signature {
+        Signature::build("touch").required(
+            "filename",
+            SyntaxShape::Path,
+            "the path of the file you want to create",
+        )
+    }
+    fn usage(&self) -> &str {
+        "creates a file"
+    }
+    fn run(
+        &self,
+        call_info: &CallInfo,
+        _registry: &CommandRegistry,
+        raw_args: &RawCommandArgs,
+        _input: Value,
+    ) -> Result<OutputStream, ShellError> {
+        call_info
+            .process(&raw_args.shell_manager, raw_args.ctrl_c.clone(), touch)?
+            .run()
+    }
+}
+fn touch(args: TouchArgs, _context: &RunnablePerItemContext) -> Result<OutputStream, ShellError> {
+    match OpenOptions::new()
```

Comment:
Can you add a test that this creates a file?

Reply:
How would one write a test for that?

Reply:
You can look through the tests we have now to get a feel for how they work. Really all you'll need to do is `touch mytestfile` and then make sure that the file exists after the test runs

Reply:
Done